### ISSUE TITLE
Cannot install browsersync on Windows 10

### ISSUE THREAD

**Issue by jelyjo** [BrowserSync/browser-sync#1396]
### Issue details

_Please provide issue details here_.

Receiving an error when attempting to install on Windows 10

_Please provide necessary steps for reproduction of this issue, or better the
reduced test case (without any external dependencies)_.

After entering npm install browser-sync the following error occurs

96 error Unexpected end of input at 1:37360
96 error .4.10":{"name":"browser-sync-ui","version":"0.4.10","dependencies":{"
96 error                                                                      ^

According to https://www.browsersync.io/docs Visual Studio should be installed and that the compilation worked fine with Visual Studio 2013 Update 4.

However the current free Visual Studio Version is Visual Studio Community 2017. After installing this I still receive the error. 

After installing Visual Studio 2017 I then also tried npm install -g browser-sync --msvs_version=2017 and I still received the same error message.

### Please specify which version of Browsersync, node and npm you're running

- Browsersync [    ]
- Node        [ v8.0.0   ]
- Npm         [ v5.0.0   ]

### Affected platforms

- [ ] linux
- [10 ] windows
- [ ] OS X
- [ ] freebsd
- [ ] solaris
- [ ] other _(please specify which)_

### Browsersync use-case

- [ ] API
- [ ] Gulp
- [ ] Grunt
- [ ] CLI

### If CLI, please paste the entire command below

{cli command here}

### for all other use-cases, (gulp, grunt etc), please show us exactly how you're using Browsersync

{Browsersync init code here}

**Comment by ccrayton** [comment]
do you have Node.js installed first? browserSync is a dependency of Node.js

**Comment by jelyjo** [comment]
Yes, I have Node v8 installed. I installed Visual Studio, no dice. I
installed the C++ workloads for Visual Studio, no dice. I installed all
workloads for Visual Studios, no dice. I am now running all individual C++
tools in Visual Studio that were not already installed to see if that will
work. The
line {"name":"browser-sync-ui","version":"0.4.10","dependencies":{ has me
thinking there is a dependency somewhere that is missing.

**Comment by mst101** [comment]
I have a similar(?) problem on Windows 10 version 1607 (build 14393.1358)
node 8.1.3
npm 5.0.4

=======================================
$ npm install -g browser-sync
npm ERR! Unexpected end of input at 1:21994
npm ERR! 1.2"},"directories":{},"dist":{"shasum":"912a831a7ef98945cceaacd7540a
npm ERR!
=======================================

Installed Visual Studio (seems excessive to have to install a whole new IDE ?) and I also installed the Visual C++ 2017 Redistributable (x64), but I still get the same error...

**Comment by jelyjo** [comment]
I gave up trying to install. I found that it is not necessary to install
Visual Studio, the redistributable packages exist so that a PC with
dependencies does not have to have the software. I downloaded and installed
all redistributable packages back to 2005 for both x64 and x86, which
believe me was a pain because the Microsoft website does not make it easy
to find them and then half the links to the packages are broken. I wasted a
day searching for and downloading these packages. I even ensured all of the
nodes listed as dependencies were installed as well, and still it would not
install. I had considered rolling back the version of node and npm I was
using but I was beyond frustrated and gave up. I was able to install
browser-sync-client, browser-sync-spa, bs-html-injector, bs-latency,
bs-recipes, bs-rewrite-rules, and bs-snippet-injector as well as all the
other dependencies but I cannot install browser-sync or browser-sync-ui.

Since I could not get the gulp process to work without the browser-sync I
eventually just set up NetBeans to compile my Sass files and I have to
remember to Ctrl+Shift+R on Google Chrome to refresh, but it still does not
enable me to see it on a smart phone which I am understanding is a big part
of the purpose of browser sync.

**Comment by mst101** [comment]
@jelyjo this is now working for me. No idea what's changed, but you might want to give it another go?

**Comment by monizza** [comment]
I was having same issue for the browser sync it got resolved by installing visual c++ redistributable and then run the installation command for browser-ync

**Comment by sakshichopkar** [comment]
![screenshot 86](https://user-images.githubusercontent.com/33214556/41462470-96d373bc-70b0-11e8-9197-83fcf69e8a3f.png)

When I am installing browser sync the following issues were generating. Help me fix it.

**Comment by rajendra-jat7** [comment]
i have
node = v12.17.0
npm = 6.14.4

**Comment by rajendra-jat7** [comment]
![Screenshot (78)](https://user-images.githubusercontent.com/65971760/84652997-2f94f580-af2a-11ea-9a40-15a5c686b80b.png)

**Comment by ritiknxt** [comment]
Hey, @rajendra-jat7  I am facing same issues as you did and am using same versions of node.js and npm. Did you problem solve?

**Comment by hussainmv** [comment]
> Hey, @rajendra-jat7 I am facing the same issues as you did and am using same versions of node.js and npm. Did you problem solve?

same problems! any idea or solution?

**Comment by ritiknxt** [comment]
Hey @hussainmv try to install an older version of browser sync. It helped in my case, hope it does in your case too.

**Comment by oftharsh** [comment]
Following are the steps to solve all these problems. Hope it helps:

1) "x packages are looking for funding." when running npm install on a react project.
     $ npm install
        packages are looking for funding.
        run `npm fund` for details.

       Solution: npm config set fund false --location=global

2)  When I run the external tool Browser Sync, I get the error:

    'browser-sync' is not recognized as an internal or external command, operable program, or batch file.

    Solution:    Go to file explorer and search for browser sync and see where the file is installed.

Then Right-click your computer, in properties, click advanced system settings, then environment variables and add the directory. For me, it was "C:\Users\userName\AppData\Roaming\npm" and save to your environment variables.
Then open the cmd and open that directory and then run the browser-sync.
Then running browser-sync --version will work. I will link a Youtube video on how to do this.

https://www.youtube.com/watch?v=bEroNNzqlF4

Make sure you edit the Path System Variable when doing so and add C:\Users\userName\AppData\Roaming\npm as a folder.
